tamr_toolbox/data_io/cloud.py

```diff
@@ -0,0 +1,220 @@
+import tempfile
+import tarfile
+
+
+def gcs_upload(
+    cloud_client,
+    source_filepath="path_to_my_local_file",
+    destination_filepath="path_to_my_file_in_bucket",
+    bucket_name="sample_bucket",
+    tar_file=True,
+    return_uploaded_file=None,
+):
+    """Upload data to a Google storage bucket
+    Args:
+        source_filepath: file path of source file being uploaded
+        destination_filepath: path to google bucket
+        cloud_client: google storage client with user credentials
+        bucket_name: name of google bucket
+        tar_file: Tar file before upload
+        return_uploaded_file: If filepath is given,
+        downloads copy of file being uploaded to filepath
+
+
+    """
+    bucket = cloud_client.get_bucket(bucket_name)
+    blob = bucket.blob(destination_filepath)
+
+    if tar_file:
+        with tempfile.NamedTemporaryFile(delete=False, mode="w") as tmp:
+            with tarfile.open(tmp.name, "w") as tar:
+                tar.add(source_filepath)
+            blob.upload_from_filename(tmp.name)
+            if return_uploaded_file:
+                with tarfile.open(return_uploaded_file, "w") as tar:
+                    tar.add(source_filepath)
+
+    else:
+        blob.upload_from_filename(source_filepath)
+        if return_uploaded_file:
+            with tarfile.open(return_uploaded_file, "w") as tar:
+                tar.add(source_filepath)
+
+
+def gcs_download(
+    cloud_client,
+    source_filepath="path_to_my_file_in_bucket",
+    destination_filepath="path_to_my_local_file",
+    bucket_name="sample_bucket",
+    untar=False,
+):
+    """ Download data to a Google storage bucket
+    Args:
+        source_filepath: Path to file being downloaded on Google
+        destination_filepath: Destination filepath for file being downloaded
+        cloud_client: google storage client with user credentials
+        bucket_name: name of google bucket being downloaded from
+        untar: Whether to decompress file being downloaded (Tar)
+    """
+
+    bucket = cloud_client.get_bucket(bucket_name)
+    this_blob = bucket.blob(source_filepath)
+
+    if untar:
+        with tempfile.NamedTemporaryFile(delete=False, mode="w") as temp_path:
+            this_blob.download_to_filename(temp_path.name)
+
+            if tarfile.is_tarfile(temp_path.name):
+                with tarfile.open(temp_path.name) as tar_file:
+                    tar_file.extractall(destination_filepath)
+            else:
+                this_blob.download_to_filename(destination_filepath)
+    else:
+        this_blob.download_to_filename(destination_filepath)
+
+
+def s3_upload(
+    cloud_client,
```

Comment:
Similar to GCS, you'll need to `from mypy_boto3_s3.client import S3Client` and type hint like this:
```suggestion
    cloud_client: S3Client,
```